azurelinuxagent/ga/exthandlers.py

```diff
@@ -969,38 +968,60 @@ def report_ext_handlers_status(self, incarnation_changed=False):
 
         self.write_ext_handlers_status_to_info_file(vm_status)
 
-    @staticmethod
-    def write_ext_handlers_status_to_info_file(vm_status):
-        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
+    def write_ext_handlers_status_to_info_file(self, vm_status):
+
+        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE.format(self.last_etag))
 
         agent_details = {
             "agent_name": AGENT_NAME,
-            "current_version": str(CURRENT_VERSION),
             "goal_state_version": str(GOAL_STATE_AGENT_VERSION),
-            "distro_details": "{0}:{1}".format(DISTRO_NAME, DISTRO_VERSION),
-            "last_successful_status_upload_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
             "python_version": "Python: {0}.{1}.{2}".format(PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO),
             "crp_supported_features": [name for name, _ in get_agent_supported_features_list_for_crp().items()],
-            "extension_supported_features": [name for name, _ in get_agent_supported_features_list_for_extensions().items()]
+            "extension_supported_features": [name for name, _ in
+                                             get_agent_supported_features_list_for_extensions().items()],
+            "last_successful_status_upload_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
         }
 
-        # Convert VMStatus class to Dict.
-        data = get_properties(vm_status)
+        if vm_status is not None:
+            # Convert VMStatus class to Dict.
+            data = get_properties(vm_status)
+
+            # .get handles the keyError and accepts a default value as 2nd parameter. If no second parameter is
+            # provided it uses None. Using {'default': None} in the case of 'vmAgent' or any other key is not available
+
+            agent_details["daemon_version"] = data.get('vmAgent', {'default': None}).get('version')
+            agent_details["distro_details"] = "{0}:{1}".format(data.get('vmAgent', {'default': None}).get('osname'),
+                                                               data.get('vmAgent', {'default': None}).get('osversion'))
+            agent_details["agent_status"] = data.get('vmAgent', {'default': None}).get('status')
+            agent_details["agent_message"] = data.get('vmAgent', {'default': None}).get('message')
+            agent_details["agent_hostname"] = data.get('vmAgent', {'default': None}).get('hostname')
+
+            goal_state_status = data.get('vmAgent', {'default': None}).get('vm_artifacts_aggregate_status',
+                                                {'default': None}).get('goal_state_aggregate_status', {'default': None})
+            goal_state_aggregate_status = {
+                "processed_time": time.strftime("%Y-%m-%dT%H:%M:%SZ",
+                                                goal_state_status.get('_GoalStateAggregateStatus__utc_timestamp')),
```

Comment:
Not sure what happens if we get None here. Quick test gave me this error, could you please double check and handle if this is true.
```
$python main.py
Traceback (most recent call last):
  File "main.py", line 6, in <module>
    print(time.strftime("%Y-%m-%dT%H:%M:%SZ", None))
TypeError: argument must be 9-item sequence, not None
```

Reply:
Good catch @nagworld9 , Addressed.